cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -125,13 +126,13 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
           .setTags(false);
       validateDefaultBranch(cmd.callAsMap(), config.getDefaultBranch());
     } catch (TransportException e) {
-      throw new RepositoryConfigValidationException("Failed to connect with remote repository: " + e.getMessage(), e);
+      throw new RemoteRepositoryValidationException("Failed to connect with remote repository: " + e.getMessage(), e);
     } catch (GitAPIException e) {
-      throw new RepositoryConfigValidationException("Failed to list remotes in remote repository: " + e.getMessage(),
+      throw new RemoteRepositoryValidationException("Failed to list remotes in remote repository: " + e.getMessage(),
```

Comment:
I don't think calling all GitAPIExceptions as server errors is good. If there is an SLO on server errors, false positives will burn the SLO.

Reply:
How do we currently handle SLO for external service interactions like source/sink? I think this should be a similar problem there also

Reply:
Changed back to `RepositoryConfigValidationException`